sdk/deviceupdate/azure-iot-deviceupdate/src/test/java/com/azure/iot/deviceupdate/DeviceManagementClientSyncStackTests.java

```diff
@@ -0,0 +1,234 @@
+        Assertions.assertEquals(200, response.iterableByPage().iterator().next().getStatusCode());
+        Assertions.assertTrue(response.stream().findAny().isPresent());
+    }
+
+    @Test
+    public void testGetDeployment() {
+        Response<BinaryData> response =
+            deviceManagementClient.getDeploymentWithResponse(TestData.DEVICE_GROUP, TestData.DEPLOYMENT_ID, requestOptions);
+        Assertions.assertEquals(200, response.getStatusCode());
+        Assertions.assertNotNull(response.getValue());
+    }
+
+    @Test
+    public void testListDeploymentsForDeviceClassSubgroup() {
+        PagedIterable<BinaryData> response =
+            deviceManagementClient.listDeploymentsForDeviceClassSubgroup(
+                TestData.DEVICE_GROUP, TestData.DEVICE_CLASS_ID, requestOptions);
+        Assertions.assertEquals(200, response.iterableByPage().iterator().next().getStatusCode());
+        Assertions.assertTrue(response.stream().findAny().isPresent());
+    }
+
+    @Test
+    public void testGetDeploymentForDeviceClassSubgroup() {
+        Response<BinaryData> response =
+            deviceManagementClient.getDeploymentForDeviceClassSubgroupWithResponse(
+                TestData.DEVICE_GROUP, TestData.DEVICE_CLASS_ID, TestData.DEPLOYMENT_ID, requestOptions);
+        Assertions.assertEquals(200, response.getStatusCode());
+        Assertions.assertNotNull(response.getValue());
+    }
+
+    @Test
+    public void testGetDeploymentStatus() {
+        Response<BinaryData> response =
+            deviceManagementClient.getDeploymentStatusWithResponse(TestData.DEVICE_GROUP, TestData.DEPLOYMENT_ID, requestOptions);
+        Assertions.assertEquals(200, response.getStatusCode());
+        Assertions.assertNotNull(response.getValue());
+    }
+
+    @Test
+    public void testGetDeviceClassSubgroupDeploymentStatus() {
+        Response<BinaryData> response =
+            deviceManagementClient.getDeviceClassSubgroupDeploymentStatusWithResponse(
+                TestData.DEVICE_GROUP, TestData.DEVICE_CLASS_ID, TestData.DEPLOYMENT_ID, requestOptions);
+        Assertions.assertEquals(200, response.getStatusCode());
+        Assertions.assertNotNull(response.getValue());
+    }
+
+    @Test
+    public void testListDeviceStatesForDeviceClassSubgroupDeployment() {
+        PagedIterable<BinaryData> response =
+            deviceManagementClient.listDeviceStatesForDeviceClassSubgroupDeployment(
+                TestData.DEVICE_GROUP, TestData.DEVICE_CLASS_ID, TestData.DEPLOYMENT_ID, requestOptions);
+        Assertions.assertEquals(200, response.iterableByPage().iterator().next().getStatusCode());
+        Assertions.assertTrue(response.stream().findAny().isPresent());
+    }
+
+    private HttpClient buildSyncAssertingClient(HttpClient httpClient) {
+        return new AssertingHttpClientBuilder(httpClient)
+            .skipRequest((ignored1, ignored2) -> false)
```

Comment:
nit: can remove this line

Reply:
Do you mean this line: .skipRequest((ignored1, ignored2) -> false)?